### PR TITLE
New Doc for Deferrable Operators on Software

### PR DESCRIPTION
**Pull request by kwcanuck**
Resolves #613.

Currently, this is a content dump from Astro with minor editing. The items @paolaperaza and @aspain indicated should be included in the topic  all appear to be present.

### CODE REVIEW

**Comment by aspain** [comment]
@kwcanuck would it make sense to provide a mention & link to https://docs.astronomer.io/software/configure-deployment#triggerer somewhere in the new Deferrable Operator section?

**Comment by kwcanuck** [comment]
@dylanbstorey can you provide the sequence/architectural diagram you requested, or provide the name of a contact who can provide it? A draft of what you're requesting would be helpful.

@sunkickr I can add the link you requested to the How It Works > Triggerer section. Would that work?

File: software/deferrable-operators.md [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+For more information on how deferrable operators work and how to use them, see the [Airflow Guide for Deferrable Operators](https://www.astronomer.io/guides/deferrable-operators) or the [Apache Airflow documentation](https://airflow.apache.org/docs/apache-airflow/stable/concepts/deferring.html).
+
+## Prerequisites
+
+To use deferrable operators both in a local Airflow environment and on Astronomer Software, you must have:
+
+- An [Astronomer Software project](create-project.md) running [Astro Runtime version 4.2.0 or later](runtime-release-notes.md#astro-runtime-420).
+- The [Astro CLI version 1.1.0 or later](https://docs.astronomer.io/astro/cli-release-notes#v110) installed.
+
+All versions of Astro Runtime version 4.2.0 and later support the Triggerer and have the `astronomer-providers` package installed. See [Astro Runtime Release Notes](runtime-release-notes.md) or [Upgrade Astro Runtime](upgrade-runtime.md).
```
Comment (jwitz) [inline software/deferrable-operators.md:43]: We need a note here about Certified versions which support deferrable operators. Customers using AC must be running at least 2.2. They also need to manually install `astronomer-providers` by adding it to the `requirements.txt` file of their Astro project.

**Review by jwitz — COMMENTED** [review]
One minor comment on some missing Software details, will defer to @aspain and @paolaperaza for final review

**Comment by sunkickr** [comment]
I think @aspain requested the link?

**Comment by kwcanuck** [comment]
> I think @aspain requested the link?

@sunkickr yes he did - sorry. Working on multiple PRs and I added the wrong reviewer name.

**Review by aspain — APPROVED** [review]

File: software/deferrable-operators.md [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+
+Deferrable operators rely on a new Airflow component called the Triggerer. The Triggerer is highly available and built into all Astronomer Software Deployments, which means that you can use deferrable operators in your DAGs with no additional configuration. To ensure that you can test your DAGs locally, the Triggerer is also built into the Astro CLI.
```
[the remark this answers is not in the document]
Comment (kwcanuck) [reply software/deferrable-operators.md:10]: @aspain I have removed the sentence from the topic.

File: software/deferrable-operators.md (hunk 1 above)
Comment (jwitz) [reply software/deferrable-operators.md:46]: Sorry should have clarified - adding `astronomer-providers` is necessary only when using a Certified image. This package is installed by default on Runtime images. 

Might be worth having a separate prerequisites bullet point list for Certified?

**Review by jwitz — APPROVED** [review]
One more comment about the Certified prerequisites, will leave it to you how to address that! Otherwise LGTM.

**Comment by kwcanuck** [comment]
The link requested by @aspain has been added and the image requested by @dylanbstorey will be added at a later date.

File: software/deferrable-operators.md [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+---
+
+[Apache Airflow 2.2](https://airflow.apache.org/blog/airflow-2.2.0/) introduced [**deferrable operators**](https://airflow.apache.org/docs/apache-airflow/stable/concepts/deferring.html), a powerful type of Airflow operator that's optimized for lower resource costs and improved performance. In Airflow, it's common to use [sensors](https://airflow.apache.org/docs/apache-airflow/stable/concepts/sensors.html) and some [operators](https://airflow.apache.org/docs/apache-airflow/stable/concepts/operators.html) to configure tasks that wait for some external condition to be met before executing or triggering another task. While tasks using standard operators and sensors take up a worker slot when checking if an external condition has been met, deferrable operators suspend themselves during that process. This releases the worker to take on other tasks.
+
+Deferrable operators rely on a new Airflow component called the Triggerer. The Triggerer is highly available and built into all Astronomer Software Deployments. To ensure that you can test your DAGs locally, the Triggerer is also built into the Astro CLI.
```
[the remark this answers is not in the document]
Comment (kwcanuck) [reply software/deferrable-operators.md:10]: @paolaperaza what is the specific edit you're requesting here?

[the remark this answers is not in the document]
Comment (kwcanuck) [reply software/deferrable-operators.md:10]: @paolaperaza what is the specific edit your are requesting?

File: software/deferrable-operators.md [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+
+For more information on how deferrable operators work and how to use them, see the [Airflow Guide for Deferrable Operators](https://www.astronomer.io/guides/deferrable-operators) or the [Apache Airflow documentation](https://airflow.apache.org/docs/apache-airflow/stable/concepts/deferring.html).
+
+## Prerequisites
+
+To use deferrable operators both in a local Airflow environment and on Astronomer Software, you must have:
+
+- An [Astronomer Software project](create-project.md) running [Astro Runtime version 4.2.0 or later](runtime-release-notes.md#astro-runtime-420).
+- The [Astro CLI version 1.1.0 or later](https://docs.astronomer.io/astro/cli-release-notes#v110) installed.
+- Astronomer Certified version 2.2 or later (if your organization is using Astronomer Certified images).
```
[the remark this answers is not in the document]
Comment (kwcanuck) [reply software/deferrable-operators.md:42]: @paolaperaza sorry, I'm not clear what specific edit you are requesting.

File: software/deferrable-operators.md [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+- The `astronomer-providers` package added to your Astro project `requirements.txt` file (if your organization is using Astronomer Certified images). This package is installed by default with Astro Runtime images. See [Add Python and OS-level Packages](https://docs.astronomer.io/software/customize-image#add-python-and-os-level-packages).
+- Triggerers enabled. See [Enable Triggerers](https://docs.astronomer.io/software/configure-deployment#enable-triggerers).
+
+All versions of Astro Runtime version 4.2.0 and later support the Triggerer and have the `astronomer-providers` package installed. See [Astro Runtime Release Notes](runtime-release-notes.md) or [Upgrade Astro Runtime](upgrade-runtime.md).
```
[the remark this answers is not in the document]
Comment (kwcanuck) [reply software/deferrable-operators.md:46]: @paolaperaza sorry, I'm not clear what specific edit you are requesting. These are separate statements and should remain.

File: software/deferrable-operators.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (paolaperaza) [reply software/deferrable-operators.md:42]: Per conversation in standup, let's somehow break up the prerequisite differences between Certified and Runtime. Either:

- Two sections, one for Runtime, one for Certified
- One section that assumes Runtime, with an intro sentence and 2 bullet points for AC-only

[the remark this answers is not in the document]
Comment (kwcanuck) [reply software/deferrable-operators.md:42]: @paolaperaza this change has been implemented.

File: software/deferrable-operators.md [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+For more information on how deferrable operators work and how to use them, see the [Airflow Guide for Deferrable Operators](https://www.astronomer.io/guides/deferrable-operators) or the [Apache Airflow documentation](https://airflow.apache.org/docs/apache-airflow/stable/concepts/deferring.html).
+
+## Prerequisites
+
+To use deferrable operators both in a local Airflow environment and on Astronomer Software, you must have:
+
+- An [Astronomer Software project](create-project.md) running [Astro Runtime version 4.2.0 or later](runtime-release-notes.md#astro-runtime-420).
+- The [Astro CLI version 1.1.0 or later](https://docs.astronomer.io/astro/cli-release-notes#v110) installed.
+- Astronomer Certified version 2.2 or later (if your organization is using Astronomer Certified images).
+- The `astronomer-providers` package added to your Astro project `requirements.txt` file (if your organization is using Astronomer Certified images). This package is installed by default with Astro Runtime images. See [Add Python and OS-level Packages](https://docs.astronomer.io/software/customize-image#add-python-and-os-level-packages).
```
[the remark this answers is not in the document]
Comment (kwcanuck) [reply software/deferrable-operators.md:43]: @paolaperaza resolved by the other edit request.

File: software/deferrable-operators.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (kwcanuck) [reply software/deferrable-operators.md:46]: @paolaperaza resolved by the other edit request.

**Pull request merged**